glean-core/ios/Glean/Net/PingUploadScheduler.swift

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// `true` to allow the uploader to process pings.
+/// Note that this does not mean that an uploader is actually running.
+/// It will be invoked when a ping is submitted.
+///
+/// `false` to stop the uploader from starting new uploads.
+/// An already running uploader will finish work and then stop.
+var stateRunAllowed: AtomicBoolean = AtomicBoolean(false)
```

Comment:
Can we avoid a global variable here? I am unsure where `shutdownUploader` and `startUploader` are used, but we should avoid global functions and variables.

Reply:
The bug mentioned does cover our intent to clean this up, we never were happy with needing to use a global to control the state here.